### PR TITLE
feat and test: Gnosis Chain bridge contract infrastructure

### PR DESCRIPTION
**Pull request by kupermind**
- Adding Gnosis Chain related set of contracts for the cross-chain bridging;
- Adding full test coverage.

### CODE REVIEW

File: .github/workflows/workflow.yaml [hunk 1]
```diff
@@ -15,6 +15,11 @@ jobs:
         with:
           submodules: recursive
 
+      - name: Get a specific Node.js version such that hardhat does not hang periodically
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18.15.0"
+
```
Comment (kupermind) [inline .github/workflows/workflow.yaml:22]: This is to fix the github workflow hang

File: contracts/bridges/HomeMediator.sol [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+/// @dev Target execution failed.
+/// @param target Target address.
+/// @param value Provided value.
+/// @param payload Provided payload.
+error TargetExecFailed(address target, uint256 value, bytes payload);
+
+/// @title HomeMediator - Smart contract for the governor home (gnosis chain) bridge implementation
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author AL
+contract HomeMediator {
```
Comment (kupermind) [inline contracts/bridges/HomeMediator.sol:46]: The contract is very similar to the fx one.

File: contracts/bridges/test/HomeMediatorTest.sol [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+/// @title HomeMediatorTest - Smart contract for the governor home (gnosis chain) bridge implementation transfer data test
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author AL
+contract HomeMediatorTest {
```
Comment (kupermind) [inline contracts/bridges/test/HomeMediatorTest.sol:7]: This is the contract to do a first simple test of the received message.

File: contracts/bridges/test/MockAMBMediator.sol [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+pragma solidity ^0.8.19;
+
+interface IHomeMediator {
+    function processMessageFromForeign(bytes memory data) external;
+}
+
+/// @title MockAMBMediator - Smart contract for mocking the AMBMediator contract on the Gnosis chain
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author AL
+contract MockAMBMediator {
```
Comment (kupermind) [inline contracts/bridges/test/MockAMBMediator.sol:11]: In tests we can't act from the deployer, and thus the AMB Proxy mock is needed.

**Review by 77ph — APPROVED** [review]

File: contracts/bridges/HomeMediator.sol [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    event FundsReceived(address indexed sender, uint256 value);
+    event ForeignGovernorUpdated(address indexed foreignMessageSender);
+    event MessageReceived(address indexed foreignMessageSender, bytes data);
+
+    // Default payload data length includes the number of bytes of at least one address (20 bytes or 160 bits),
+    // value (12 bytes or 96 bits) and the payload size (4 bytes or 32 bits)
+    uint256 public constant DEFAULT_DATA_LENGTH = 36;
+    // FX child address on L2 that receives the message across the bridge from the foreign L1 network
```
Comment (mariapiamo) [inline contracts/bridges/HomeMediator.sol:54]:
```suggestion
    // Home address on L2 that receives the message across the bridge from the foreign L1 network
```

File: README.md [hunk 6]
```diff
@@ -169,3 +179,4 @@ The governance contracts and the rest was inspired and based on the following so
 The bridging contracts were based on and inspired by the following sources:
 - [Polygon](https://github.com/maticnetwork).
 - [fx-portal](https://github.com/fx-portal).
+- [Gnosis Chain Docs](https://docs.gnosischain.com/).
```
[the remark this answers is not in the document]
Comment (kupermind) [reply README.md:182]: Fixed in the next PR

File: README.md [hunk 7]
```diff
@@ -33,9 +33,12 @@ The changelog leading to the implementation of `wveOLAS` can be found here: [Cha
 
 To complement, a list of known vulnerabilities can be found here: [Vulnerabilities list #1](https://github.com/valory-xyz/autonolas-governance/blob/main/docs/Vulnerabilities_list%231.pdf?raw=true)
 
-In order to manage cross-bridge transactions via the `Timelock` contract, the Fx Governor Tunnel contract is implemented:
+In order to manage cross-bridge transactions via the `Timelock` contract on the polygon network, the Fx Governor Tunnel contract is implemented:
```
[the remark this answers is not in the document]
Comment (kupermind) [reply README.md:36]: Fixed in the last PR

File: README.md [hunk 8]
```diff
@@ -33,9 +33,12 @@ The changelog leading to the implementation of `wveOLAS` can be found here: [Cha
...
 - [FxGovernorTunnel](https://github.com/valory-xyz/autonolas-governance/blob/main/contracts/bridges/FxGovernorTunnel.sol).
 
+In order to manage cross-bridge transactions via the `Timelock` contract on the gnosis network, the Home Mediator contract is implemented:
```
[the remark this answers is not in the document]
Comment (kupermind) [reply README.md:39]: Fixed in the last PR

**Review by mariapiamo — APPROVED** [review]

**Pull request merged**